src/map/guild.cpp

```diff
@@ -330,6 +330,92 @@ uint64 CastleDatabase::parseBodyNode(const YAML::Node &node) {
 		safestrncpy(gc->castle_event, npc_name.c_str(), sizeof(gc->castle_event));
 	}
 
+	if (this->nodeExists(node, "SiegeType")) {
+		std::string siege_type;
+
+		if (!this->asString(node, "SiegeType", siege_type))
+			return 0;
+
+		std::string type_constant = "SIEGE_TYPE_" + siege_type;
+		int64 constant;
+
+		if (!script_get_constant(type_constant.c_str(), &constant)) {
+			this->invalidWarning(node["SiegeType"], "SiegeType %s doesn't exist, skipping.\n", type_constant.c_str());
+			return 0;
+		}
+
+		gc->siege_type = static_cast<e_siege_type>(constant);
+	}
+	else {
+		if (!exists)
+			gc->siege_type = SIEGE_TYPE_TE;
+	}
+
+
+	if (this->nodeExists(node, "EnableClientWarp")) {
+		bool enable_client_warp;
+
+		if (!this->asBool(node, "EnableClientWarp", enable_client_warp))
+			return 0;
+
+		gc->enable_client_warp = enable_client_warp;
+	}
+	else {
+		if (!exists)
+			gc->enable_client_warp = true;
+	}
+
+	if (this->nodeExists(node, "WarpX")) {
+		uint16 warp_x;
+
+		if (!this->asUInt16(node, "WarpX", warp_x)) {
+			this->invalidWarning(node["WarpX"], "Invalid WarpX %hu, defaulting to 0.\n", warp_x);
+			gc->warp_x = 0;
+		}
+
+		gc->warp_x = warp_x;
+	}
```

Comment:
Isn't a default value of `gc->warp_x` required if `EnableClientWarp` is true and `WarpX` is missing?

Reply:
No.
How we can check this while parsing yaml?

Reply:
How about just add
```
		gc->warp_x = warp_x;
	}
	else {
		if (!exists)
			gc->warp_x = 0;
	}
```
(if you do like that don't forget warp_y and to update the doc)

Reply:
Another way could be to check if warpx and y exist in enableclientwarp, if the castle entry doesn't exist and enableclientwarp is true